### ISSUE TITLE
active learning data pruning

### ISSUE THREAD

**Issue by anhhuyalex** [anhhuyalex/mind-reader#2]
jan21_2321_pruning: filter out 8500 images / 9000 in cache

jan22_2221_pruning_1000:
- random 8000 sets
- pruned from best 8500 images: start with 
`if  [int(i) for i in samples_list[1:100].split(" ") if i != ""] != [70420, 2598, 63069, 32343, 37942, 25934, 32988, 15835, 50918, 48882, 28161, 13467, 6373, 1247, 41268, 66508, 31]: continue`

# generally jobs look like this
jan22_2221_pruning_${NUM_SAMPLES}_from_${TOTAL_SAMPLES}
jan22_2221_pruning_1500_from_8000 vs. jan22_2221_pruning_1500_from_9000
jan22_2221_pruning_2000_from_7500 vs. jan22_2221_pruning_2000_from_9000
jan22_2221_pruning_2500_from_7000 vs. jan22_2221_pruning_2500_from_9000
jan22_2221_pruning_3000_from_6500 vs. jan22_2221_pruning_3000_from_9000
jan22_2221_pruning_3500_from_6000 vs. jan22_2221_pruning_3500_from_9000
vs. jan22_2221_pruning_8500_from_9000

forward pruning (adding data samples from small runs instead of taking out data samples)

**Comment by anhhuyalex** [comment]
comparison of using 750 images provided
baseline of using the first 750 trials from NSD
brain clustering 
gaudy / contrastiness and look at pixel values: 255 values on the pixels, look at the mean value (127: put to maximum)
metric for gaudiness -- look up some measure of contrast of images

**Comment by anhhuyalex** [comment]
for i in 0 1 2 3 4 5 6 7 8 9; do /weka/home-alexnguyen/mind-reader/MindEyeV2/src/mindeye/bin/accelerate launch --num_processes=$(($NUM_GPUS * $COUNT_NODE)) --num_machines=$COUNT_NODE --main_process_ip=$MASTER_ADDR --main_process_port=$MASTER_PORT --mixed_precision=fp16 \
    kmeans_v2.py --data_path=${DATA_PATH} \
    --model_name=${model_name} --no-multi_subject --subj=1 \
    --batch_size=28 --num_sessions=40 \
    --hidden_dim=1024 --clip_scale=1. \
    --no-blurry_recon --blur_scale=.5  \
    --seq_past=0 --seq_future=0 \
    --cache_dir=/weka/home-alexnguyen \
    --no-use_prior --prior_scale=30 \
    --n_blocks=4 --max_lr=3e-4 --mixup_pct=.33 --num_epochs=30 --no-use_image_aug \
    --ckpt_interval=1 --no-ckpt_saving --wandb_log --wandb_group_name=mar11_2345_clipembds_kmeans_pruned_ids_nclusters_750_normalized_True  \
    --seed=$i \
    --multisubject_ckpt=./datasets--pscotti--mindeyev2/snapshots/183269ab73b49d2fa10b5bfe077194992934e4e6/train_logs/multisubject_subj01_1024hid_nolow_300ep \
    --is_keep_only_new_imagepresentations \
    --filter_samples=/weka/home-alexnguyen/mind-reader/MindEyeV2/src/cache/clipembds_kmeans_pruned_ids_nclusters_750_normalized_True_N_${NUM_SAMPLES}_seed_${i}.npy; done

for i in 0 1 2 3 4 5 6 7 8 9; do /weka/home-alexnguyen/mind-reader/MindEyeV2/src/mindeye/bin/accelerate launch --num_processes=$(($NUM_GPUS * $COUNT_NODE)) --num_machines=$COUNT_NODE --main_process_ip=$MASTER_ADDR --main_process_port=$MASTER_PORT --mixed_precision=fp16 \
    kmeans_v2.py --data_path=${DATA_PATH} \
    --model_name=${model_name} --no-multi_subject --subj=1 \
    --batch_size=28 --num_sessions=40 \
    --hidden_dim=1024 --clip_scale=1. \
    --no-blurry_recon --blur_scale=.5  \
    --seq_past=0 --seq_future=0 \
    --cache_dir=/weka/home-alexnguyen \
    --no-use_prior --prior_scale=30 \
    --n_blocks=4 --max_lr=3e-4 --mixup_pct=.33 --num_epochs=30 --no-use_image_aug \
    --ckpt_interval=1 --no-ckpt_saving --wandb_log --wandb_group_name=mar11_2345_clipembds_kmeans_pruned_ids_nclusters_750_normalized_False  \
    --seed=$i \
    --multisubject_ckpt=./datasets--pscotti--mindeyev2/snapshots/183269ab73b49d2fa10b5bfe077194992934e4e6/train_logs/multisubject_subj01_1024hid_nolow_300ep \
    --is_keep_only_new_imagepresentations \
    --filter_samples=/weka/home-alexnguyen/mind-reader/MindEyeV2/src/cache/clipembds_kmeans_pruned_ids_nclusters_750_normalized_False_N_${NUM_SAMPLES}_seed_${i}.npy; done

**Comment by anhhuyalex** [comment]
-- look if we have (714, 714) images to send to lizzie
-- send lizzie more diverse sets of images with different seeds
-- first look through wanjia + stylegan images & remove semantic duplicates, then add clustering images (while removing semantic duplicates from wanjia + stylegan)

**Comment by anhhuyalex** [comment]
-- request 2 nodes, 8 gpus per node in a job

**Comment by anhhuyalex** [comment]
retrain coreset fmri model and plot metrics
